packages/patterns/src/keys/checkKey.js

```diff
@@ -430,7 +430,7 @@ export const getCopyMapEntries = m => {
   const {
     payload: { keys, values },
   } = m;
-  const { length } = keys;
+  const { length } = /** @type {Array} */ (keys);
```

Comment:
can the param be typed? Why isn't `keys` inferred from CopyMap? Oh, it's an alias for `CopyTagged` which says `payload` is just Passable/any.  Might be a good time to have a proper CopyMap type.

Reply:
I'm abstain from whether now is a good time. But it would indeed be wonderful to have proper typing of CopyMap<K,V>, CopySet<K>, and CopyBag<K>. The other main category of Tagged are the Matchers. Would also be nice to have proper typing of them, but less important.

Reply:
Looks like @gibson042 landed a working type yesterday: https://github.com/endojs/endo/pull/1493/files#diff-a81b0292a3e5e26fea27548664397a190fdf3b8f407031f508fa1c460dce77efR112